channel_test.go

```diff
@@ -112,3 +113,18 @@ func TestIsolatedSubChannelsDontSharePeers(t *testing.T) {
 	assert.NotNil(t, sub.peers.peersByHostPort["127.0.0.1:3000"])
 	assert.Nil(t, isolatedSub.peers.peersByHostPort["127.0.0.1:3000"])
 }
+
+func TestChannelTracerMethod(t *testing.T) {
+	tracer := mocktracer.New()
+	ch, err := NewChannel("svc", &ChannelOptions{
+		Tracer: tracer,
+	})
+	require.NoError(t, err)
+	defer ch.Close()
+	assert.Equal(t, tracer, ch.Tracer(), "expecting tracer passed at initialization")
+
+	ch, err = NewChannel("svc", &ChannelOptions{})
+	require.NoError(t, err)
+	defer ch.Close()
+	assert.EqualValues(t, opentracing.GlobalTracer(), ch.Tracer(), "expecting default tracer")
```

Comment:
can you change the opentracing global tracer, and make sure that it's returned when you call `ch.Tracer()` (basically testing that we're calling the function every time instead of caching it).